ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -1164,6 +1154,7 @@ export const WorkspaceEdit = fp.flow(
     async saveWorkspace() {
       try {
         this.setState({ loading: true });
+        let accessLevel = null;
```

Comment:
hm.  Looking into this a bit, I think we can remove these `let`s - which is generally something I prefer to do, because reusing mutable variables can obscure detail.

Here, we have 3 uses of `workspace`:
* lines 1158 to 1161.  We can remove this because it gets immediately overwritten by ...
* the workspace set in 1164, 1166, or 1168.  This `workspace` is used by navigate() but then is overwritten *again* by...
* the call to getWorkspace() in 1194.

My preference would be to use `const` in 1194 for `workspace` and `accessLevel`, and use a different `const` for the workspace in 1164.  What do you think?

Reply:
I can see that 1158-1161 is not necessary.

I am curious why 1174 and 1194 are necessary.  If we already have workspace, from 1164, 1166, and 1168, then why do we need to get it again? Is it about the access level?

Reply:
I don't know if it is necessary or not - there's a comment asking for that investigation so my guess is that it's more complex than it might first seem.  So it may not be worth doing that now.

Reply:
Good point. I updated the variables. What do you think?

Reply:
I'd prefer to remve the `let` at 1157 too

Reply:
I might be missing something, but I don't understand how that's possible. If 1191 refers to `workspace` and that can be set on 1160, 1162, or 1164, how could I get rid of the declaration of that variable ahead of time?

Reply:
I am also realizing that when I replaced this:
```        
        let workspace = this.state.workspace;
        if (!this.state.populationChecked) {
          workspace.researchPurpose.populationDetails = [];
        }
```
        
from 1191, the test ['should clear all selected specific populations if NO underrepresented populations study is selected' started failing](https://app.circleci.com/pipelines/github/all-of-us/workbench/32522/workflows/4fae5465-11f3-4289-b520-8210a91e91ad/jobs/351443) in workspace-edit.spec.tsx. My thinking is that `let workspace = this.state.workspace` was aliasing workspace to this.state.workspace which meant that in 
```
            workspace = await workspacesApi().updateWorkspace(
            this.state.workspace.namespace,
            this.state.workspace.id,
            { workspace: this.state.workspace }
          );
```

populationDetails was being cleared in our api.

Reply:
yuck - we should not be setting state like that.

OK - I'm convinced that this is a bigger mess than we should try to tackle in this PR.

Reply:
Agreed on both counts. Sounds like there is some further cleanup needed here.

Thanks for walking through it with me.